rules/bad_defer.go

```diff
@@ -0,0 +1,58 @@
+package rules
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/securego/gosec"
+)
+
+type badDefer struct {
+	gosec.MetaData
+	typ     string
+	methods []string
+}
+
+func (r *badDefer) ID() string {
+	return r.MetaData.ID
+}
+
+func normalize(typ string) string {
+	return strings.TrimPrefix(typ, "*")
+}
+
+func contains(methods []string, method string) bool {
+	for _, m := range methods {
+		if m == method {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *badDefer) Match(n ast.Node, c *gosec.Context) (*gosec.Issue, error) {
+	if deferStmt, ok := n.(*ast.DeferStmt); ok {
+		if typ, method, err := gosec.GetCallInfo(deferStmt.Call, c); err == nil {
+			if normalize(typ) == r.typ && contains(r.methods, method) {
+				return gosec.NewIssue(c, n, r.ID(), r.What, r.Severity, r.Confidence), nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+// NewDeferredClosing detects unsafe defer of os.Close
+func NewDeferredClosing(id string, conf gosec.Config) (gosec.Rule, []ast.Node) {
+	return &badDefer{
+		typ:     "os.File",
+		methods: []string{"Close"},
+		MetaData: gosec.MetaData{
+			ID:         id,
+			Severity:   gosec.Medium,
+			Confidence: gosec.High,
+			What:       fmt.Sprintf("Defering closing a file."),
```

Comment:
I wondering if this message shouldn't be a bit more generic something like:
```
Deferring unsafe method %q on type %q
```

and then wen the issue is created,  you can fill up the method and type.

Reply:
Changes made.